drivers/serial/uart_mcux_lpuart.c

```diff
@@ -865,6 +865,8 @@ static void mcux_lpuart_isr(const struct device *dev)
 	if (data->callback) {
 		data->callback(dev, data->cb_data);
 	}
+
+	LPUART_ClearStatusFlags(config->base, kLPUART_RxOverrunFlag);
```

Comment:
Q: do you need to clear OR when only CONFIG_UART_ASYNC_API is enabled?

Reply:
Well... I was not sure about that case TBH and didn't test that.

Reply:
Thank you for this change, Please enclose this so this is done only when there is an overrun
`if ((uint32_t)kLPUART_RxOverrunFlag == ((uint32_t)kLPUART_RxOverrunFlag & status))`

Reply:
@mmahadevan108 Done as requested.